homeassistant/components/light/tplink.py

```diff
@@ -119,30 +125,51 @@ def rgb_color(self):
 
     @property
     def is_on(self):
-        """True if device is on."""
+        """Return True if device is on."""
         return self._state
 
     def update(self):
         """Update the TP-Link Bulb's state."""
         from pyHS100 import SmartDeviceException
         try:
+            if self._supported_features == 0:
+                self.get_features()
             self._state = (
                 self.smartbulb.state == self.smartbulb.BULB_STATE_ON)
-            self._brightness = brightness_from_percentage(
-                self.smartbulb.brightness)
-            if self.smartbulb.is_color:
+            if self._name is None:
+                self._name = self.smartbulb.alias
+            if self.smartbulb.is_dimmable:
```

Comment:
This is already known after calling `get_features()`, so checking against your cached value would avoid doing further I/O here. See also other `is_*` checks below related to this.